### ISSUE TITLE
Scalar Binomial support validation allows y outside [0,1]; Bernoulli-logit likelihood becomes unbounded

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#451]
## Summary

Scalar `Binomial` support validation allows invalid responses. For $y\notin[0,1]$ the Bernoulli-logit likelihood is unbounded above, so the GLM objective is not well-posed.

**Where:**

* `src/types.rs`, around lines 364–437 and 479–507.
* `src/solver/workflow.rs`, around lines 3102–3121 and 5358–5387.
* `src/solver/pirls/mod.rs`, around lines 6220–6258 and 6386–6395.

The scalar `Binomial` family is routed as a Bernoulli/binomial-logit family, but response support validation gives `Binomial` no support requirement. Degeneracy checks only catch all-zero/all-one cases and otherwise defer to a supposed downstream binarity check. In the explicit-family standard path, there is no adequate support enforcement before PIRLS likelihood math.

The log-likelihood used is

$$\ell(\eta) = y\log\mu + (1-y)\log(1-\mu),\qquad \mu=\sigma(\eta).$$

Using the logit identity this is $\ell(\eta) = y\eta - \log(1+e^\eta)$.

For a Bernoulli response, $y$ must be 0 or 1. For a soft-label cross-entropy interpretation, at minimum $0\le y\le 1$ is required.

If the code accepts $y=2$, then $\ell(\eta) = 2\eta-\log(1+e^\eta)$. As $\eta\to\infty$, $\log(1+e^\eta)\sim \eta$, so $\ell(\eta)\sim \eta\to\infty$. The likelihood is unbounded above — not a valid Bernoulli/binomial likelihood.

The deviance implementation also becomes meaningless for invalid $y$: it contains terms $(1-y)\log\frac{1-y}{1-\mu}$, which assume $1-y\ge0$. For $y>1$ the binomial deviance is outside its domain.

## Correct version

If `ResponseFamily::Binomial` means Bernoulli, validate exactly $y_i\in\{0,1\}$:

```rust
Self::Binomial => yi.is_finite()
    && ((yi - 0.0).abs() <= tol || (yi - 1.0).abs() <= tol)
```

Then the all-zero/all-one degeneracy check remains meaningful.

If soft labels are intentionally allowed, document the family as Bernoulli cross-entropy / quasi-binomial soft-label mode, with support $0\le y_i\le 1$.

If true binomial counts are intended, the model needs trial counts $m_i$, with

$$\ell_i = y_i\log\mu_i + (m_i-y_i)\log(1-\mu_i) + \log\binom{m_i}{y_i},\qquad y_i\in\{0,\dots,m_i\}.$$

The current scalar path has no such $m_i$, so permitting arbitrary finite values is mathematically wrong.

**Comment by SauersML** [comment]
Fixed in 8e182651e (Round 1). Scalar Binomial support is now validated (Bernoulli y∈{0,1} / documented soft-label [0,1]) before PIRLS. Adversary-approved; CI verifies.
